File: codeforces/71A_waytoolongowrds.py
n = int(input())
words = []
for i in range(n):
    words.append(input())

for i in range(n):
    word = words[i]
    abbr = ""
    l = len(word)
    if l > 10:
        abbr = abbr + word[0]
        abbr = abbr + str(l - 2)
        abbr = abbr + word[-1]
    else:
        abbr = word
    print(abbr)

File: codeforces/71A_waytoolongowrds.cpp
#include <bits/stdc++.h>
using namespace std;

int main(){
    int n, l, abbr_num;
    cin >> n;
    string word, abbr, result[n];
    for(int i=0; i < n; i++){
        cin >> word;
        l = word.length();
        if(l > 10){
                    abbr.push_back(word[0]);
            cout << abbr << "\n";
            abbr_num = word.length() - 2;
            abbr.push_back(abbr_num);
            abbr.push_back(word.back());
            result.push_back(abbr);
        }
        else{
            result.push_back(word);
        }
    }
    cout << result << "\n";
}